eZ/Publish/Core/Persistence/Cache/UrlAliasHandler.php

```diff
@@ -44,7 +44,10 @@ public function publishUrlAliasForLocation(
                 'alwaysAvailable' => $alwaysAvailable,
             )
         );
-        $this->clearLocation($locationId);
```

Comment:
Shouldn't we make it more general and also consider other uses of `clearLocation` method? It seems that the aforementioned function will cause issues in all cases where URL changes and the location has children. So, also in {{locationSwapped}} and possibly {{translationRemoved}}?

Reply:
good point

Reply:
I have checked and the other uses of `clearLocation` method are not affected:
* `locationSwapped` is not affected because the children are not moved alongside the parents and are not redirected (at all, so the old URL for the child returns 404 after the swap).
* `translationRemoved` is not affected because the old translated URLs return 404 after the translation is removed and are not redirected.
* `locationDeleted` is not affected because the content's additional locations have no effect on the children's URLs, so removing the location for the parent doesn't change anything for the children.